### PR TITLE
Acceptance Suite: Page Object Conversion

### PR DESCRIPTION
**Pull request by leviwilson**
This pull replaces the use of the `web_steps` that are provided with Capybara with more of a page-object approach. The gem that I've pulled in is called [site_prism](https://github.com/natritmeyer/site_prism) as it is one of the only libraries I found that supported capybara-webkit.

I took some liberties with the library as the only accessor it exposed was `element`, so I've added a [few of my own](https://github.com/leviwilson/ollert/blob/page_object_example/test/lib/core_ext/accessors.rb). So far the only additional ones I've added are:

*  `button`
* `checkbox`
* `link`
* `text_field`
* `label`

These are really just thin-wrappers around the existing SitePrism `element` accessor. As the need arises, these are easy to extend.

The goal of this PR is to simplify / clean up the step definitions to make the barrier for entry easier for others to add more acceptance tests as features are added.

Feedback welcome!

### CODE REVIEW

File: test/features/support/hooks.rb [hunk 1]
```diff
@@ -1,5 +1,12 @@
 Before do
   Mongoid.purge!
+  on(LandingPage).load
+end
+
+Before('@test_user') do
+  user = User.new
```
Comment (larryprice) [inline test/features/support/hooks.rb]: can this just be `user.create {email: Environment.test_username}`?

Comment (leviwilson) [inline test/features/support/hooks.rb]: :+1: I think I just lifted from what it was doing, but I think that's a better thing to do.

File: test/lib/navigation.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+require 'page_navigation'
+
+module OllertTest
+  module Navigation
+    include PageNavigation
+
+    def on(cls, &block)
+      screen = cls.new
+      block.call screen if block
```
Comment (larryprice) [inline test/lib/navigation.rb]: i think i'd prefer to see an `unless block.nil?` here instead of relying on truthiness.

**Comment by larryprice** [comment]
This is a fantastic PR, btw.

File: test/lib/navigation.rb (hunk 2 above)
Comment (leviwilson) [inline test/lib/navigation.rb]: 6 of 1 1/2 dozen of another :) Either way is fine with me. Would it be better if I said `if block_given?` instead?

File: test/lib/pages/settings_page.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  def delete_account
+    choose_to_delete_account
+    wait_until { no_longer_deleting? }
+  end
+
+  alias_method :orig_connect, :connect_with_a_different_account
+  def connect_with_a_different_account(&block)
+    orig_connect
+    block.call(self) if block
```
Comment (larryprice) [inline test/lib/pages/settings_page.rb]: same comment as before in favor of `block_given?`. do a quick search for any other instances.

**Comment by leviwilson** [comment]
updated the other `if block` call to use the alternative syntax

**Pull request merged**